packages/nextjs/src/utils/withSentry.ts

```diff
@@ -10,11 +10,21 @@ import {
   stripUrlQueryAndFragment,
 } from '@sentry/utils';
 import * as domain from 'domain';
-import { NextApiHandler, NextApiRequest, NextApiResponse } from 'next';
-
-// This is the same as the `NextApiHandler` type, except instead of having a return type of `void | Promise<void>`, it's
-// only `Promise<void>`, because wrapped handlers are always async
-export type WrappedNextApiHandler = (req: NextApiRequest, res: NextApiResponse) => Promise<void>;
+import { NextApiRequest, NextApiResponse } from 'next';
+
+// These are the same as the official `NextApiHandler` type, except
+//
+// a) Instead of having a return type of `void | Promise<void>` (Next < 12.1.6) or `unknown | Promise<unknown>` (Next
+// 12.1.6+), the unwrapped version has both.
+//
+// b) The wrapped version returns only promises, because wrapped handlers are always async.
+//
+// TODO: Fix the need for the multi-version hack. See https://github.com/getsentry/sentry-javascript/pull/5737.
```

Comment:
Is this really a hack? :D There will be a point where the types diverge too much between Next.js versions, so vendoring our own version-agnostic Next.js types seems rather inevitable to me. It's just nice to have the official Next.js types available to us until we can't use them anymore.

Reply:
I kinda it is, because it's really just to compensate for our dev dependency being out of date. Though I suppose that once we upgrade to Next 12, test apps built with lower versions will start complaining... Okay, so maybe it's gotta stay. I'll update the comment.